### ISSUE TITLE
My client sockets disconnect with a "ping timeout" reason after a few hours of connection and don't connect again

### ISSUE THREAD

**Issue by maximilien0405** [socketio/socket.io#5012]
Hi, i'm having a bit of a issue with my project. I'm building an app that will stay opened at all times in the device, with a non-stop connection through the sockets. However lately, all of my sockets disconnect and can't reconnect. I managed to print some logs from the device and here it is : 

![Capture d’écran 2024-04-26 à 17 07 56](https://github.com/socketio/socket.io/assets/46009279/80f09df1-42d1-4d4f-b78a-563611884d88)

As you can see on the screenshot, the socket connection starts at around 1:30 AM and then is disconnectd at 7:30 AM, but manages to reconnect each time until 9:14 where it disconnects and then can't reconnect with the reason "ping timeout".

I'm not sure where is the problem here. I'm using Angular 17 with **ngx-socket-io** version **4.6.1**. In my backend i use NestJS with **socket-io 4.7.2**. 

Since this problem that happend this morning i did theses steps to try to fix the problem : 
- Upgraded my NestJS from v9 (a year old) to v10 (including the @nestjs/websockets package)
- Upgraded socket-io in my NestJS from 4.7.2 to 4.7.5
- Added a pingTimeout of 60000 to my WebSocketGateway (maybe this will fix it ??)

Any idea what could be causing the problem ? Also it's important to note than in my APP i have an interval that tries to connect to the socket every minute just in case it's disconnected, but as you can see on the logs, it can't even reconnect 🤔 
@darrachequesne

**Comment by darrachequesne** [comment]
Hi! That's weird indeed. It seems the client is not even able to reach the server.

Is the server reachable? What happens if you restart the server? nginx? You mention using a device, are you on Android? iOS?

**Comment by maximilien0405** [comment]
Yes the server is reachable. If i restart the server nothing changes, they still can't connect, even though they try to reconnect. It happens on IOS and Android. All of my devices are turned on and the app is open h24
Right now i just added a catch to the "ping timeout" error and to a location.reload() to reload the app and reconnect to the server (hopefully). I will also implement a way to do a location.reload() once a day, just to reconnect to the socket freshly and prevent any timeout problem @darrachequesne

**Comment by Mpratyush54** [comment]
Once try that if you are emitting some random data at fix time interval probably less than 1/2 hrs as it appears that there is a time out error also try connecting it with new devices after this error arises

**Comment by GautamJain133** [comment]
yes, i am also facing the same issue if anyone knows how to solve this issue please revert. my application is always open on desktop either in minimize or in focus. but after some random time. on server it is showing that user is disconencted due to pingTimeout. i am using the latest version of socket io client and socket io both. website is build using mern

**Comment by GautamJain133** [comment]
@maximilien0405 have u got any permanent solution rather that the hacks like window.reload and all. is this window.reload mechanism works properly

**Comment by GautamJain133** [comment]
@darrachequesne please help

**Comment by maximilien0405** [comment]
@GautamJain133 Yes the window.reload works but like you said it's not a great solution.. but also i think it's nice sometime to reload the socket connection if it's connected for long periods of time, so until a solution is found it's not that bad :/

**Comment by mainisameer** [comment]
Folks we are facing the same issue on Chrome where this ping pong mechanism is causing a whole bunch of issues. Is there a way to disable this ping pong

**Comment by darrachequesne** [comment]
Hi! Were you able to find the culprit?
